lib/ansible/modules/network/f5/bigip_command.py

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2016 F5 Networks Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: bigip_command
+short_description: Run arbitrary command on F5 devices.
+description:
+  - Sends an arbitrary command to an BIG-IP node and returns the results
+    read from the device. This module includes an argument that will cause
+    the module to wait for a specific condition before returning or timing
+    out if the condition is not met.
+version_added: "2.4"
+options:
+  commands:
+    description:
+      - The commands to send to the remote BIG-IP device over the
+        configured provider. The resulting output from the command
+        is returned. If the I(wait_for) argument is provided, the
+        module is not returned until the condition is satisfied or
+        the number of retires as expired.
+      - The I(commands) argument also accepts an alternative form
+        that allows for complex values that specify the command
+        to run and the output format to return. This can be done
+        on a command by command basis. The complex argument supports
+        the keywords C(command) and C(output) where C(command) is the
+        command to run and C(output) is 'text' or 'one-line'.
+    required: True
+  wait_for:
+    description:
+      - Specifies what to evaluate from the output of the command
+        and what conditionals to apply.  This argument will cause
+        the task to wait for a particular conditional to be true
+        before moving forward. If the conditional is not true
+        by the configured retries, the task fails. See examples.
+    required: False
+    default: None
```

Comment:
@Akasurde understood that it's not required, I've left them in for the other F5 modules because I copy paste the doc block around a lot when I need to write unit and integration tests. So I use it as a reference then and when I am answering customer questions. Does that suffice?